Add Amazon S3 scenario doc (#1238)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/amazons3-connector/amazons3-connector-example.md

# Amazon S3 Connector Example 

> The AmazonS3 Connector allows you to access the REST API of Amazon Simple Storage Service (Amazon S3), which provides a simple web services interface that can be used to store and retrieve any amount of data, at any time, from anywhere on the web. It gives any developer access to the same highly scalable, reliable, secure, fast, inexpensive infrastructure that Amazon S3 uses to run its own global network of websites. The service aims to maximize benefits of scale and to pass on those benefits to developers.

Reviewer: The first sentence is a bit too similar to what we have in the AWS docs. Please reword this if possible.

Author: Updated.

Hence WSO2 EI AmazonS3 Connector is useful to take your on-premise data to the cloud. The advantage is, you do not need to worry about managing and replicating data on-premise. 

## What you'll build

> This example depicts you how to use AmazonS3 connector to 

Reviewer: ```suggestion
This example depicts how to use AmazonS3 connector to:
```

Author: Updated.

> 1. Create a S3 bucket (a location for storing your data) in Amazon cloud 

Reviewer: Add a period at the end of each bullet point or list number.

Author: Updated.

2. Upload a message into the created bucket as a text file
3. Retrieve created text file back and convert into a message in WSO2 EI integration. 

All three operations are exposed via an API. The API with the context `/s3connector` has three resources  

* `/createbucket` - Once invoked, it will create a bucket in Amazon with the specified name
* `/addobject`  - The incoming message will be stored into the specified bucket with the specified name
* `/file/{filename}` - Once invoked, it will read the specified file from the specified bucket and respond with the content of the file

> <img src="/assets/img/connectors/Amazon-s3-diagram.png" title="Overview of Amazon S3 use case" width="800" alt="Amazon S3 use case"/>

Reviewer: We need to explain this a bit more. It's not clear how all those three actions are represented by that first arrow and we need to go through the flow and explain a bit. Some improvements to the diagram might also be needed to enhance the clarity here.

Author: Updated.

## Setting up the environment 

Please follow the steps mentioned at [Setting up AmazonS3](<link>) document in order to create a Amazon S3 account and obtain
credentials you need to access the Amazon APIs. Keep them saved to be used in the next steps.  

## Configure the connector in WSO2 Integration Studio

Follow these steps to set up the ESB Solution Project and import AmazonS3 connector into it.

{!references/connectors/importing-connector-to-integration-studio.md!} 

1. Right click on the created ESB Solution Project and select, -> **New** -> **Rest API** to create the REST API.
   <img src="/assets/img/connectors/adding-an-api.png" title="Adding a Rest API" width="800" alt="Adding a Rest API"/>

> 2. Specify API name as `S3ConnectorTestAPI` and API context as `/s3connector`. You can go to the source view of the xml configuration file of the API and copy the following configuration.

Reviewer: ```suggestion
2. Specify the API name as `S3ConnectorTestAPI` and API context as `/s3connector`. You can go to the source view of the XML configuration file of the API and copy the following configuration.
```

Author: Updated.

    </addMessage>
    ```
2. Invoke the API as shown below using the curl command. Curl Application can be downloaded from [here] (https://curl.haxx.se/download.html).
    ```
    curl -H "Content-Type: application/xml" --request POST --data @data.xml http://127.0.0.1:8290/s3connector/addobject
    ```
**Expected Response**:
    You will receive a response like below containing the details of the object created.  
    ```
    <?xml version="1.0" encoding="UTF-8"?>
    <CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
        <Location>http://wso2engineers.s3.amazonaws.com/Julian.txt</Location>
        <Bucket>wso2engineers</Bucket>
        <Key>Julian.txt</Key>
        <ETag>&quot;2b492c33895569c5c06cd7942f42914f-1&quot;</ETag>
    </CompleteMultipartUploadResult>
    ```
    Navigate to AWS S3 console and click on the bucket `wso2engineers`. You will note that a file has been created with the name `Julian.txt`. 
    <img src="/assets/img/connectors/AmazonS3-bucket-upload.png" title="Upload object to Amazon S3 bucket" width="800" alt="Upload object to Amazon S3 bucket"/>

### Read objects from Amazon S3 bucket

> Now let us read the information on the WSO2 engineer we stored on the Amazon S3 bucket.

Reviewer: ```suggestion
Now let us read the information on `wso2engineers` that we stored in the Amazon S3 bucket.
```

Author: Updated.

1. Create a file called data.xml with the following content. It specifies which bucket to read from and what the filename is. This example assumes that the object is stored at root level inside the bucket. You can also read a object stored in a folder inside the bucket. 
    ```
    <getObject>
    <objectName>Julian.txt</objectName>
    <bucketName>wso2engineers</bucketName>
    </getObject>
    ```
2. Invoke the API as shown below using the curl command.  
    ```
    curl -H "Content-Type: application/xml" --request POST --data @data.xml http://127.0.0.1:8290/s3connector/info
    ```
**Expected Response**:
    You will receive a response like below containing the details of the engineer requested. 
    ```
    Julian Garfield, Software Engineer, Integration Group
    ```

> In this example Amazon S3 connector is used to perform operations with Amazon S3 storage. Upon the errors occur when invoking S3 operations you can receive details of the error by the S3 responses itself. Please read on Amazon connector and the reference guide to know more operations you can perform with Amazon S3 connector. 

Reviewer: ```suggestion
In this example Amazon S3 connector is used to perform operations with Amazon S3 storage. You can receive details of the errors that occur when invoking S3 operations using the S3 responses itself. Please read the Amazon S3 connector reference guide to learn more about the operations you can perform with the Amazon S3 connector. 
```

Author: Updated.
